lib/common/database.dart

```diff
@@ -7,18 +7,15 @@ import 'package:planner/models/task.dart';
 
 class DatabaseService {
   static final DatabaseService _singleton = DatabaseService._internal();
-  static late String userid;
+  late String userid;
 
   // TODO: Add caching layer here if time permits
 
   // users collection reference
   late CollectionReference users =
   FirebaseFirestore.instance.collection('users');
 
-  factory DatabaseService({String? uid}) {
-    if (uid != null) {
-      userid = uid;
-    }
+  factory DatabaseService() {
```

Comment:
can't we still pass a string in here and call initUID? Or is not allowed by dart?

Reply:
dart doesn't allow referencing instance members even if we don't grab the value itself.
![image](https://github.com/Scrumbags115/plannertarium/assets/122345910/5a5b8a50-f111-49f3-a637-25e00cfc3b81)
it's suggestion is literally the equivalent of FU